File: projects/test_models/Zeta/cs/Types/CollectionTestType.cs
using System;

namespace Phi
{
    /// <summary>
    /// Test type
    /// </summary>
    public sealed class CollectionTestType
    {
        #region Properties
        /// <summary>
        /// Some comment
        /// </summary>
        public System.Collections.Generic.List<object> BinariesTest { get; set; }
        #endregion

        #region Constructors
        public CollectionTestType() { }
        public CollectionTestType(System.Collections.Generic.List<object> binariesTest)
        {
            BinariesTest = binariesTest;
        }
        #endregion

        #region Equality
        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != GetType()) return false;

            var value = obj as CollectionTestType;
            if (value == null) return false;

            return
                BinariesTest != null && value.BinariesTest != null &&
                BinariesTest.Equals(value.BinariesTest);
        }

        public static bool operator ==(CollectionTestType lhs, CollectionTestType rhs)
        {
            if (Object.ReferenceEquals(lhs, rhs))
                return true;

            return !Object.ReferenceEquals(null, lhs) && lhs.Equals(rhs);
        }

        public static bool operator !=(CollectionTestType lhs, CollectionTestType rhs)
        {
            return !(lhs == rhs);
        }

        public override int GetHashCode()
        {
            unchecked
            {
                // Choose large primes to avoid hashing collisions
                const int HashingBase = (int) 2166136261;
                const int HashingMultiplier = 16777619;

                int hash = HashingBase;
                hash = (hash * HashingMultiplier) ^
                    (!object.ReferenceEquals(null, BinariesTest) ? BinariesTest.GetHashCode() : 0);
                return hash;
            }
        }
        #endregion
    }
}

File: projects/test_models/Zeta/cpp/src/types/CollectionTestType.cpp
#include "Phi/types/CollectionTestType.hpp"

namespace Phi {

CollectionTestType::CollectionTestType(const std::vector<std::string>& BinariesTest)
    : BinariesTest_(BinariesTest) { }

void CollectionTestType::swap(CollectionTestType& other) noexcept {
    using std::swap;
    swap(BinariesTest_, other.BinariesTest_);
}

bool CollectionTestType::operator==(const CollectionTestType& rhs) const {
    return BinariesTest_ == rhs.BinariesTest_;
}

CollectionTestType& CollectionTestType::operator=(CollectionTestType other) {
    using std::swap;
    swap(*this, other);
    return *this;
}

const std::vector<std::string>& CollectionTestType::BinariesTest() const {
    return BinariesTest_;
}

std::vector<std::string>& CollectionTestType::BinariesTest() {
    return BinariesTest_;
}

void CollectionTestType::BinariesTest(const std::vector<std::string>& v) {
    BinariesTest_ = v;
}

void CollectionTestType::BinariesTest(const std::vector<std::string>&& v) {
    BinariesTest_ = std::move(v);
}

}
